packages/qwik/src/optimizer/src/plugins/vite.ts

```diff
@@ -1119,30 +1119,48 @@ function absolutePathAwareJoin(path: Path, ...segments: string[]): string {
 export function convertManifestToBundleGraph(manifest: QwikManifest): QwikBundleGraph {
   const bundleGraph: QwikBundleGraph = [];
   const graph = manifest.bundles;
-  const map = new Map<string, { index: number; deps: string[] }>();
-  for (const bundleName in graph) {
+  if (!graph) {
+    return [];
+  }
+  const names = Object.keys(graph).sort();
+  const map = new Map<string, { index: number; deps: Set<string> }>();
+  const clearTransitiveDeps = (deps: Set<string>, seen: Set<string>, bundleName: string) => {
+    const bundle = graph[bundleName];
+    for (const dep of bundle.imports || []) {
+      if (deps.has(dep)) {
```

Comment:
the name `deps` here is confusing, I would rename it to `parentDeps` and `parentBundle` so it'll be easier to follow.

(or you could change the `dep` in the loop to `childDep` )